src/plist.h

```diff
@@ -291,6 +291,54 @@ static inline Obj PopPlist(Obj list)
     return val;
 }
 
+
+/****************************************************************************
+**
+*F  NewEmptyPlist() . . . . . . . . . . . . . . create a new empty plain list
+*F  ImmutableEmptyPlist . . . . . . . . . . . . an immutable empty plain list
+**
+**  ImmutableEmptyPlist is a variable rather than a function, as we can
+**  reuse the same immutable empty plist.
+*/
+static inline Obj NewEmptyPlist(void)
+{
+    return NEW_PLIST(T_PLIST_EMPTY, 0);
+}
+
+extern Obj ImmutableEmptyPlist;
+
+/****************************************************************************
+**
+*F  NewPlistFromArray(<list>,<length>) . . create a plain list from a C array
+*/
+static inline Obj NewPlistFromArray(const Obj * list, Int length)
+{
+    if (length == 0) {
+        return NewEmptyPlist();
+    }
+
+    Obj o = NEW_PLIST(T_PLIST, length);
+    SET_LEN_PLIST(o, length);
+    memcpy(BASE_PTR_PLIST(o), list, length * sizeof(Obj));
+    return o;
+}
+
+/****************************************************************************
+**
+*F  NewPlistFromArgs(<args...>) .  create a plain list from list of arguments
+**
+**  This macro turns a variable-length list of macro arguments into an array,
+**  which is then passed to NewPlistFromArray.
+**
+**  __VA_ARGS__ contains the list of arguments given to this macro.
+**  (Obj[]){ __VA_ARGS__ } creates an array of Obj containing the elements
+**  of __VA_ARGS__.
+**  sizeof(A)/sizeof(Obj) gives us the number of elements in the array A.
```

Comment:
Minor nitpick, no change needed: This would be self-explanatory if you used  the `ARRAY_SIZE` macro, wouldn't it?

Reply:
Hadn't noticed we had one :)